### PR TITLE
Make compiler configurable part2

### PR DESCRIPTION
**Pull request by SrodriguezO**
Provide default compilers for various versions of Scala and Play and make the default used configurable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ Simple Core API: [play_routes](docs/stardoc/play-routes.md)
 For more information about the Play Framework, see [the Play documentation](https://www.playframework.com/documentation/latest).
 
 ## Installation
-Create a file called at the top of your repository named `WORKSPACE` and add the following snippet to it.
+Create a file at the top of your repository named `WORKSPACE` and add the following snippet to it.
 
-```python
```
Comment (borkaehw) [inline README.md]: I like the python highlight, why removing it :(

File: README.md [hunk 2]
```diff
@@ -35,12 +35,41 @@ http_archive(
...
+
+We provide 5 default compilers:
+
+- For Scala 2.11 + Play 2.5: `@io_bazel_rules_play_routes//default-compiler-clis:scala_2_11_play_2_5`
+- For Scala 2.11 + Play 2.6: `@io_bazel_rules_play_routes//default-compiler-clis:scala_2_11_play_2_6`
+- For Scala 2.11 + Play 2.7: `@io_bazel_rules_play_routes//default-compiler-clis:scala_2_11_play_2_7`
+- For Scala 2.12 + Play 2.6: `@io_bazel_rules_play_routes//default-compiler-clis:scala_2_12_play_2_6`
+- For Scala 2.12 + Play 2.7: `@io_bazel_rules_play_routes//default-compiler-clis:scala_2_12_play_2_7`
+
+To bind one of the default compilers, simply specify the correct Play version in the call to `play_routes_repositories` and update the bind statement:
```
Comment (borkaehw) [inline README.md:61]: Would it be better to make it clear Scala 2.11 and 2.12 will be provided once Play 2.6 or 2.7 is specified? Basically let user know why there is no config for Scala version.

File: README.md [hunk 3]
```diff
@@ -35,12 +35,41 @@ http_archive(
...
+```
+play_routes_repositories(<Play Version>)
 load("@play_routes//:defs.bzl", play_routes_pinned_maven_install = "pinned_maven_install")
 play_routes_pinned_maven_install()
+
+bind(
+  name = "default-play-routes-compiler-cli",
+  actual = <default compiler Label>
```
Comment (borkaehw) [inline README.md]: `<Default Compiler Label>` maybe?

File: workspace.bzl [hunk 4]
```diff
@@ -4,23 +4,80 @@ Load 3rd party maven dependencies
...
-            "org.scala-sbt:util-interface:1.2.0",
-            "org.scala-lang:scala-compiler:2.11.12",
-            "org.scala-lang:scala-library:2.11.12",
-            "org.scala-lang:scala-reflect:2.11.12",
+        artifacts = play_artifacts + common_artifacts,
+        repositories = [
+            "http://central.maven.org/maven2",
+        ],
+        fetch_sources = True,
+        maven_install_json = maven_install_json,
```
Comment (borkaehw) [inline workspace.bzl]: How about this
```
maven_install_json = "@io_bazel_rules_play_routes//:play_{}_routes_compiler_cli_install.json".format(play_version.replace(".", "_"))
```

File: workspace.bzl [hunk 5]
```diff
@@ -4,23 +4,80 @@ Load 3rd party maven dependencies
...
-            "com.github.scopt:scopt_2.11:3.7.0",
-            "com.lucidchart:play-routes-compiler-cli:0.1",
-            "com.typesafe.play:routes-compiler_2.11:2.5.19",
-            "org.scala-sbt:zinc_2.11:1.2.1",
-            "org.scala-sbt:compiler-interface:1.2.1",
-            "org.scala-sbt:util-interface:1.2.0",
-            "org.scala-lang:scala-compiler:2.11.12",
-            "org.scala-lang:scala-library:2.11.12",
-            "org.scala-lang:scala-reflect:2.11.12",
+        artifacts = play_artifacts + common_artifacts,
```
Comment (borkaehw) [inline workspace.bzl]: How about this
```
artifacts = play_artifacts[play_version] + common_artifacts,
```

File: workspace.bzl [hunk 6]
```diff
@@ -4,23 +4,80 @@ Load 3rd party maven dependencies
 
 load("@rules_jvm_external//:defs.bzl", "maven_install")
 
-def play_routes_repositories():
+def play_routes_repositories(play_version):
+
+    play_2_5_artifacts = [
```
Comment (borkaehw) [inline workspace.bzl]: How about this
```
play_artifacts = {
    "2.5": [
        "com.lucidchart:play-routes-compiler-cli_2.11:2.5.19",
    ],
    "2.6": [
        "com.lucidchart:play-routes-compiler-cli_2.11:2.6.23",
        "com.lucidchart:play-routes-compiler-cli_2.12:2.6.23",
    ],
    "2.7": [
        "com.lucidchart:play-routes-compiler-cli_2.11:2.7.2",
        "com.lucidchart:play-routes-compiler-cli_2.12:2.7.3",
    ]
}
```

File: README.md (hunk 1 above)
Comment (SrodriguezO) [reply README.md]: I'll bring it back :)

File: default-compiler-clis/BUILD.bazel [hunk 7]
```diff
@@ -0,0 +1,34 @@
+java_binary(
```
Comment (borkaehw) [inline default-compiler-clis/BUILD.bazel:1]: We could do something like this
```
scala_play_version = [
    ["2_11", "2_5"],
    ["2_11", "2_6"],
    ["2_11", "2_7"],
    ["2_12", "2_6"],
    ["2_12", "2_7"]
]
[
    java_binary(
      name = "scala_{}_play_{}".format(scala_version, play_version),
      runtime_deps = ["@play_routes//:com_lucidchart_play_routes_compiler_cli_{}".format(scala_version)],
      main_class = "rulesplayroutes.routes.CommandLinePlayRoutesCompiler",
      visibility = ["//visibility:public"]
    )
    for scala_version, play_version in scala_play_version
]
```
You would also want to verify the syntax.

File: workspace.bzl [hunk 8]
```diff
@@ -4,23 +4,76 @@ Load 3rd party maven dependencies
...
-            "org.scala-lang:scala-reflect:2.11.12",
+        artifacts = play_artifacts[play_version] + common_artifacts,
+        repositories = [
+            "http://central.maven.org/maven2",
+        ],
+        fetch_sources = True,
+        maven_install_json = "@io_bazel_rules_play_routes//:play_{}_routes_compiler_cli_install.json".format(play_version.replace(".", "_"))
+    )
+
+    maven_install(
```
Comment (borkaehw) [inline workspace.bzl]: Also something like this
```
play_version = ["2_5", "2_6", "2_7"]
[
    maven_install(
        name = "play_{}_routes_compiler_cli".format(version),
        artifacts = play_artifacts[version] + common_artifacts,
        repositories = [
            "http://central.maven.org/maven2",
        ],
        fetch_sources = True,
        maven_install_json = "@io_bazel_rules_play_routes//:play_{}_routes_compiler_cli_install.json".format(version),
    )
    for version in play_version
]
```

**Review by borkaehw — APPROVED** [review]
It would be good to simplify code furthermore, otherwise LGTM.

File: default-compiler-clis/BUILD.bazel (hunk 7 above)
Comment (SrodriguezO) [reply default-compiler-clis/BUILD.bazel:1]: I don't like the idea of using `for` loops in BUILD files.. the fact that it needs to be in a for comprehension to get around `for` loops not being allowed at the top level in BUILD files feels like a red flag

File: workspace.bzl (hunk 8 above)
Comment (SrodriguezO) [reply workspace.bzl]: This one I'm more open to. (btw I don't think this needs to be in a comprehension; I'll try it out)

File: default-compiler-clis/BUILD.bazel (hunk 7 above)
Comment (SrodriguezO) [reply default-compiler-clis/BUILD.bazel:1]: I do like that it's much more concise... On the other hand, however, this file will not be changing frequently

File: workspace.bzl (hunk 8 above)
Comment (SrodriguezO) [reply workspace.bzl]: Worked out nicely; updated

**Pull request merged**